base/abstractarray.jl

```diff
@@ -305,9 +319,12 @@ end
 
 copy(a::AbstractArray) = copy!(similar(a), a)
 
-function copy!{R,S}(B::AbstractVecOrMat{R}, ir_dest::Range{Int}, jr_dest::Range{Int}, A::AbstractVecOrMat{S}, ir_src::Range{Int}, jr_src::Range{Int})
+function copy!{R,S}(B::AbstractVecOrMat{R}, ir_dest::Range{Int}, jr_dest::Range{Int},
+                    A::AbstractVecOrMat{S}, ir_src::Range{Int}, jr_src::Range{Int})
     if length(ir_dest) != length(ir_src) || length(jr_dest) != length(jr_src)
-        throw(ArgumentError("source and destination must have same size"))
+        length(ir_dest) != length(ir_src) &&
+            throw(ArgumentError("source ($(length(ir_src))) and destination ($(length(ir_dest))) must have same size"))
+        throw(ArgumentError("source ($(length(jr_src))) and destination ($(length(jr_dest))) must have same size"))
```

Comment:
I think a common pattern is to write `source and destination must have the same size (got X and Y)`